sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Redistribute.java

```diff
@@ -160,6 +161,7 @@ public PCollection<T> expand(PCollection<T> input) {
       return input
           .apply("Pair with random key", ParDo.of(new AssignShardFn<>(numBuckets)))
           .apply(Redistribute.<Integer, T>byKey().withAllowDuplicates(this.allowDuplicates))
+          .apply("Redistribute", ParDo.of(new RedistributeFn<>()))
```

Comment:
Thanks for proposing the fix. As this is a Spark runner issue, a proper fix should be in runners/spark code path. Java core is a common code path affects all runners.

Reply:
Sure, I just move it to the right location, can you please check the code again?